### PR TITLE
Created Example for Locale API using Reactjs. Fixes issue #546.

### PR DESCRIPTION
**Pull request by k-guy-dev**
Create a simple example of using Locale API.

### CODE REVIEW

**Comment by christyjacob4** [comment]
Hi, sorry for the delay. Reviewing it now

**Comment by k-guy-dev** [comment]
Thanks @christyjacob4  😄

**Comment by christyjacob4** [comment]
I can see both a `package-lock.json` and a `yarn.lock` . Since the README uses `npm` you can consider deleting the `yarn.lock`

**Comment by k-guy-dev** [comment]
> I can see both a `package-lock.json` and a `yarn.lock` . Since the README uses `npm` you can consider deleting the `yarn.lock`

ok, I'll remove the yarn.lock

**Comment by christyjacob4** [comment]
Looks like you're using `node-appwrite` which is our server side SDK. 
You should be using `appwrite` for this client side example. 
Please update the examples using the `appwrite` npm package

**Comment by k-guy-dev** [comment]
@christyjacob4 changed from node-sdk to appwrite.

**Comment by k-guy-dev** [comment]
Please for the time being, @spielers @christyjacob4  can you add hactoberfest-accepted tag? Thanks

File: README.md [hunk 1]
```diff
@@ -6,7 +6,11 @@ This repository has various example projects created by the Appwrite community u
 
 - [**Login and Signup Page**](/example-auth/) Implementation of Appwrite's Create Session and Create User API Using ReactJS as the UI Framework.
 - [**CRUD application with Appwrite**](/example-crud/) Implementation of Appwrite's CRUD API Using ReactJS as the UI Framework.
+- [**Locale API- Getting user's Location Example**](/example/locale/) Implementation of Appwrite's Local-API using Reactjs as the UI Framework.
+<<<<<<< HEAD
 
+=======
+>>>>>>> dfb404d4ffbe0d4ba82cec7b4c36a1389a87ae8f
```
Comment (christyjacob4) [inline README.md]: can you get rid of these

File: example-locale/src/config/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const Appwrite = require("appwrite"); //Import appwrite 
+
+let sdk  = new Appwrite();
+sdk
+  .setEndpoint("http://localhost:443/v1") // Your API Endpoint
+  .setProject("[PROJECT ID]") // Your project ID
+  .setKey("[SECRET API KEY]"); // Your secret API key
```
Comment (christyjacob4) [inline example-locale/src/config/index.js]: We dont require a key for the client side sdk.

**Comment by christyjacob4** [comment]
Hey @eldadfux there seems to be a CORS issue with this demo, when I test on local 
Screenshot 2020-10-16 at 4 20 07 AM

I do get back a response on Postman but it seems like the default locale values. 
Screenshot 2020-10-16 at 4 21 19 AM

**Comment by k-guy-dev** [comment]
> Hey @eldadfux there seems to be a CORS issue with this demo, when I test on local
> Screenshot 2020-10-16 at 4 20 07 AM
> 
> I do get back a response on Postman but it seems like the default locale values.
> Screenshot 2020-10-16 at 4 21 19 AM

--------------------------------------
@christyjacob4 this is because of the new CORS policy. Make sure that the port on which the client side is running is set same as the API endpoint's port. It worked for me that way.

File: example-locale/src/config/index.js (hunk 2 above)
Comment (k-guy-dev) [reply example-locale/src/config/index.js]: Made the changes.

File: README.md (hunk 1 above)
Comment (k-guy-dev) [reply README.md]: Made the changes.

**Comment by christyjacob4** [comment]
Screenshot 2020-10-17 at 11 34 32 AM


Great work @k-guy-dev

File: example-locale/src/config/index.js (hunk 2 above)
Comment (christyjacob4) [reply example-locale/src/config/index.js]: The key field is still present ?

**Comment by k-guy-dev** [comment]
> Great work @k-guy-dev
@christyjacob4 thanks 😄

File: example-locale/src/config/index.js (hunk 2 above)
Comment (christyjacob4) [reply example-locale/src/config/index.js]:
```suggestion
```

File: example-locale/README.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+export {response};
+```
+
+A deeper inspection of this code can be found in the comments within it,
+
+TL:DR: Create a appwrite instance and initalise it with the endpoint and ProjectID of the project we are working with then export this for usage outside of the file.
+
+## Creating App.js
+
+Inside our `src/App.js` we'll be importing our two components `components/Header/js` and `components/YourInfo.js`. Header.js is just our logo and heading, where as YourInfo.js accepts the reponse of the Locale-API and displays it in the DOM.
```
Comment (christyjacob4) [inline example-locale/README.md]:
```suggestion
Inside our `src/App.js` we'll be importing our two components `components/Header.js` and `components/YourInfo.js`. Header.js is just our logo and heading, where as YourInfo.js accepts the reponse of the Locale-API and displays it in the DOM.
```

File: example-locale/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+export default App;
+```
+
+Inside our App.js we have just imported our two components. Now it's time to create YourInfo.js
+
+## Creating YourInfo.js
+
+This file recieves all the json data from the Local API. We first import the response which is basically `sdk.get();` from our index.js file. Then we use a promise to fetch the json data.
```
Comment (christyjacob4) [inline example-locale/README.md]:
```suggestion
This file recieves all the json data from the Locale API. We first import the response which is basically `sdk.get();` from our index.js file. Then we use a promise to fetch the json data.
```

File: example-locale/README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+We are using functional component in this file and we'll flood our state inside this promise only.
```
Comment (christyjacob4) [inline example-locale/README.md]: Do you think `flood our state` is the right terminology?

**Comment by christyjacob4** [comment]
@k-guy-dev Please resolve the comments and we're good to go.

File: example-locale/README.md (hunk 5 above)
Comment (k-guy-dev) [reply example-locale/README.md]: Ya, maybe we can change this to, "set our state inside this promise".

**Comment by k-guy-dev** [comment]
> @k-guy-dev Please resolve the comments and we're good to go.

Done. @christyjacob4

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
Awesome Great job

**Comment by k-guy-dev** [comment]
@eldadfux I think we can merge it now.

**Comment by spielers** [comment]
Hey, @k-guy-dev  Thanks for your contribution will get you merged!

best,
spielers

**Pull request merged**